frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.unit.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { getIcon } from "__support__/ui";
+import { createMockQueryFoxExpressions } from "metabase/query_builder/tests/common";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import ExpressionWidget, { ExpressionWidgetProps } from "./ExpressionWidget";
+
+describe("ExpressionWidget", () => {
+  it("should render proper controls", () => {
+    setup({});
+
+    expect(screen.getByText("Expression")).toBeInTheDocument();
+    expect(screen.getByText("Name")).toBeInTheDocument();
+
+    expect(screen.getByText("Cancel")).toBeInTheDocument();
+    expect(screen.getByText("Done")).toBeInTheDocument();
+  });
+
+  it("should render help icon with tooltip", () => {
+    setup({});
+
+    const icon = getIcon("info");
+    expect(icon).toBeInTheDocument();
+    userEvent.hover(icon);
+
+    userEvent.hover(icon);
+
+    expect(
+      screen.getByText(
+        "You can reference columns here in functions or equations, like: floor([Price] - [Discount]).",
+      ),
+    ).toBeInTheDocument();
+  });
+});
+
+const onChangeExpressionMock = jest.fn();
+function setup({
+  name,
+  expression,
+  query = createMockQueryFoxExpressions() as StructuredQuery,
```

Comment:
Seems like we're calling `metabase-lib/metadata/Table's` `query` method in `createMockQueryFoxExpressions`
AFAIR, it always returns a structured query, can we avoid type-casting?

Reply:
Yeah, `Table.prototype.query` should always return a `StructuredQuery`. Maybe we should explicitly specify the return types in `Table's` method instead, so we don't need to cast types elsewhere